src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Build/SolutionBuildManager.cs

```diff
@@ -168,5 +168,13 @@ public void StartUpdateSpecificProjectConfigurations(IVsHierarchy[] hierarchy, u
                 dwFlags: dwFlags,
                 fSuppressUI: 0));
         }
+
+        public int StartSimpleUpdateProjectConfiguration(IVsHierarchy pIVsHierarchyToBuild, IVsHierarchy? pIVsHierarchyDependent, string? pszDependentConfigurationCanonicalName, uint dwFlags, uint dwDefQueryResults, int fSuppressUI)
+        {
+            Assumes.NotNull(_vsSolutionBuildManager2);
```

Comment:
This can throw as the code in this PR currently stands.

Today, `_vsSolutionBuildManager2` is only non-null if something calls `SubscribeSolutionEventsAsync` first. You might be getting lucky, with another component getting in first, but it's not guaranteed.

`SolutionBuildManager` was originally written as an abstraction over SBM events, making it easier to use them correctly.

Given you don't care about events here, I wonder if you should avoid using this abstraction and just use `IVsService<SVsSolutionBuildManager, IVsSolutionBuildManager2> solutionBuildManagerService` directly instead, then call `StartUpdateSpecificProjectConfigurations` on that directly in this class.

Reply:
Updated to use the first option, also looks like we can revert the change in `SolutionBuildManager` .